lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.cpp

```diff
@@ -157,6 +158,261 @@ TypeSystemSwiftTypeRef::CanonicalizeSugar(swift::Demangle::Demangler &dem,
   });
 }
 
+swift::Demangle::ManglingErrorOr<std::string>
+TypeSystemSwiftTypeRef::TransformModuleName(
+    llvm::StringRef mangled_name,
+    const llvm::StringMap<llvm::StringRef> &module_name_map) {
+  swift::Demangle::Demangler dem;
+  auto *node = dem.demangleSymbol(mangled_name);
+  auto *adjusted_node = TypeSystemSwiftTypeRef::Transform(
+      dem, node, [&](swift::Demangle::NodePointer node) {
+        if (node->getKind() == Node::Kind::Module) {
+          auto module_name = node->getText();
+          if (module_name_map.contains(module_name)) {
+            auto real_name = module_name_map.lookup(module_name);
+            auto *adjusted_module_node =
+                dem.createNodeWithAllocatedText(Node::Kind::Module, real_name);
+            return adjusted_module_node;
+          }
+        }
+
+        return node;
+      });
+
+  auto mangling = mangleNode(adjusted_node);
+  return mangling;
+}
+
+static NodePointer CloneNode(Demangler &dem, NodePointer node,
+                             bool set_children = true) {
+  NodePointer clone;
+  auto kind = node->getKind();
+  if (node->hasText())
+    clone = dem.createNodeWithAllocatedText(kind, node->getText());
+  else if (node->hasIndex())
+    clone = dem.createNode(kind, node->getIndex());
+  else
+    clone = dem.createNode(kind);
+  if (set_children)
+    for (NodePointer transformed_child : *node)
+      clone->addChild(transformed_child, dem);
+  return clone;
+}
+
+NodePointer TypeSystemSwiftTypeRef::TransformTupleElements(
```

Comment:
How is this different from the `Transform()` function?

Reply:
This only applies the transformation function to the elements of a tuple. It _could_ be implemented in terms of `Transform`, but I think it'd suffer in terms of legibility, since the implementation would be less straightforward.